src/widget/wtrackmenu.cpp

```diff
@@ -1595,6 +1625,161 @@ void WTrackMenu::slotClearAllMetadata() {
             &trackOperator);
 }
 
+namespace {
+
+class RemoveTrackFilesFromDiskTrackPointerOperation : public mixxx::TrackPointerOperation {
+  public:
+    mutable QList<TrackRef> tr_tracksToPurge;
```

Comment:
We have no agreed on a `tr_` prefix and know one will know that it is an abbreviation for TrackRef -> m_tracksToPurge/m_tracksToKeep

Make those members private and add public member functions tracksToPurge()/tracksToKeep() for accesing the outcome of the batch operation. Also add a comment why mutating operations on the `TrackModel` have to be deferred.

Reply:
:+1: 

I like "know one will know" : )